presto-docs/src/main/sphinx/develop/client-protocol.rst

```diff
@@ -0,0 +1,164 @@
+===========================
+Presto Client HTTP Protocol
+===========================
+
+The Presto client allows users to submit Presto queries and view results.  This note documents the protocol
+used by the Presto client.
+
+HTTP Methods
+============
+
+* A ``POST`` to ``/v1/statement`` runs the query string in the ``POST`` body, and returns a JSON document containing
+  the query results.  If there are more results, the JSON document will contain a ``nextUri``
+  URL attribute.
+* A ``GET`` to the ``nextUri`` attribute returns the next batch of query results.
+* A ``DELETE`` to ``nextUri`` terminates a running query.
+
+Overview of Query Processing
+============================
+
+A Presto client request is initiated by an HTTP ``POST`` to the endpoint ``/v1/statement``, with a ``POST`` body
+consisting of the SQL query string.  The caller may set header ``X-Presto-User`` to the username for the session,
+as well as a long list of other headers, documented below.
+
+If the client request returns an HTTP 503, that means the server was busy, and the client should try again
+in 50-100 millisecons.  Any HTTP status other than 503 or 200 means that the query has failed.
+
+The ``/v1/statement`` ``POST`` request returns a JSON document of type ``QueryResults``, as well as a collection
+of response headers.  The ``QueryResults`` document will contain an ``error`` field of type ``QueryError``
+if the query has failed, and if that object is not present, the query succeeded.
+
+If the ``data`` field of the JSON document is set, it contains a list of the rows
+of data, and the ``columns`` field will also be set to a list of the names and types of
+the columns returned by the query.  Most of the response headers should be treated like
+browser cookies by the client, and echoed back as request headers is subsequent client requests,
+as documented below.
+
+If the JSON document returned by the ``POST`` to ``/v1/statement`` does not contain a ``nextUri`` link, the query has completed,
+either successfully or unsuccessfully, and no additional requests need be made.  If the ``nextUri`` link is present in
```

Comment:
```suggestion
either successfully or unsuccessfully, and no additional requests need to be made.  If the ``nextUri`` link is present in
```

Reply:
Changed as suggested, since perhaps the subjunctive is a bit confusion.